### PR TITLE
CFE-380 - Improvements related to security audit

### PR DESCRIPTION
**Pull request by sherine-k**
- Enables gosec in .golangci.yaml
- Annotate all false positives if any
- Fix G306, mainly on test files, putting all os.WriteFile permissions to 0600 instead of 0644
- Introduce the usage of kubelet-serving-ca.crt when calling Kubelet on HTTPS (enable TLS check)

**Should not be merged before fixing the G402 - TLS InsecureSkipVerify set true**

### CODE REVIEW

File: cmd/node-observability-agent/main.go [hunk 1]
```diff
@@ -59,16 +60,22 @@ func main() {
...
 		Token:          token,
+		CACerts:        caCerts,
 		StorageFolder:  *storageFolder,
 		CrioUnixSocket: *crioSocket,
 		NodeIP:         node,
 	})
 }
 
-func checkParameters(tokenFile string, nodeIP string, storageFolder string, crioUnixSocket string) {
+func checkParameters(tokenFile string, nodeIP string, storageFolder string, crioUnixSocket string, caCertFile string) {
```
Comment (lmzuccarelli) [inline cmd/node-observability-agent/main.go:77]: This is great - good one

**Comment by lmzuccarelli** [comment]
@sherine-k @thejasn - could you take a look at the Makefile. It seems the  `make generate` recipe has been removed.

**Comment by thejasn** [comment]
> @sherine-k @thejasn - could you take a look at the Makefile. It seems the `make generate` recipe has been removed.


Looks like it never had `make generate`, this [PR](https://github.com/openshift/node-observability-agent/pull/26) fixes the issue.

**Pull request merged**